components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -606,17 +608,40 @@ export default class PluginManagement extends AdminSettings {
         return this.installFromUrl(true);
     }
 
+    handleRemovePluginCancel = () => {
+        this.setState({
+            showRemoveModal: false,
+        });
+        this.resolveRemoveModal(false);
+    }
+
+    handleRemovePlugin = () => {
```

Comment:
To keep the flow a bit simpler here, we could have the remove button do the `showRemovePluginModal` [here](https://github.com/mattermost/mattermost-webapp/blob/a6074923f8ac42f3c058b1801a7f3ca98358783a/components/admin_console/plugin_management/plugin_management.jsx#L793) and then `handleRemovePlugin` would call `handleRemove` directly:

```suggestion
    handleRemovePlugin = () => {
        this.setState({showRemoveModal: false});
        this.handleRemove();
    }
```
This way we can keep all of the remove logic in the original `handleRemove` and we do not need to do check the result of the `showRemovePluginModal` given that it will trigger the appropriate handler as needed

Reply:
I agree, it took me a second to figure out what was going on with the promise handling. Clever yet a bit confusing.

Also agree with making `setState` a one-liner.

Reply:
If we do that the local state has to be changed to add the pluginId in it(because `this.handleRemove()` will not know which plugin to remove). The flow will be: 
1. user clicks remove
2. we add pluginid to the state
3. show modal
4. if ok get pluginid and remove plugin
5. delete pluginid from state

Promise basically does the same thing. I like it :))

Reply:
The pluginId is already being set in the state [here](https://github.com/mattermost/mattermost-webapp/blob/9f755c4e11dacf2cc044b7baccf7ed2a2e08cb7b/components/admin_console/plugin_management/plugin_management.jsx#L637) so it would be to move this up to this `handleRemovePlugin` and null it out in the cancel case: 

```suggestion
    handleRemovePlugin = (e) => {
         pluginId = e.currentTarget.getAttribute('data-plugin-id');
         this.setState({showRemoveModal: false, removing: pluginId});
         this.handleRemove();
    }
```

Agreed that the promise is another way, but given that it was a bit confusing already for me and @mickmister, I'm 3/5 on changing it.

Reply:
Yes, having the function receive the plugin id through a parameter would be ideal. I would prefer the parameter be the actual plugin id instead of the the `event/data-plugin-id` combo. Passing a lambda into the `PluginItem` component's prop is technically frowned upon like so:

```
handleRemove={() => this.handleRemove(pluginStatus.id)}
```

But the code reads more clearly and it doesn't require introducing an html level attribute.

Reply:
The reason it's passed in as an event data is because it will now be the handler of the button remove click so that's the way it determines which plugin was clicked as it is currently doing at [here](https://github.com/mattermost/mattermost-webapp/blob/9f755c4e11dacf2cc044b7baccf7ed2a2e08cb7b/components/admin_console/plugin_management/plugin_management.jsx#L633)